tests/integration/setting_test.go

```diff
@@ -55,36 +55,41 @@ func TestSettingShowUserEmailProfile(t *testing.T) {
 	req := NewRequest(t, "GET", "/user1")
 	resp := session.MakeRequest(t, req, http.StatusOK)
 	htmlDoc := NewHTMLParser(t, resp.Body)
+	assert.NotContains(t, htmlDoc.doc.Find(".user.profile").Text(), "user1@noreply.example.org")
 	assert.Contains(t, htmlDoc.doc.Find(".user.profile").Text(), "user1@example.com")
 
-	// user1 can not see user2's hidden email
+	// user1 can not see user2's hidden email (or their corresponding noreply email)
 	req = NewRequest(t, "GET", "/user2")
 	resp = session.MakeRequest(t, req, http.StatusOK)
 	htmlDoc = NewHTMLParser(t, resp.Body)
 	// Should only contain if the user visits their own profile page
+	assert.NotContains(t, htmlDoc.doc.Find(".user.profile").Text(), "user2@noreply.example.org")
 	assert.NotContains(t, htmlDoc.doc.Find(".user.profile").Text(), "user2@example.com")
 
 	// user2 can see user1's visible email
 	session = loginUser(t, "user2")
 	req = NewRequest(t, "GET", "/user1")
 	resp = session.MakeRequest(t, req, http.StatusOK)
 	htmlDoc = NewHTMLParser(t, resp.Body)
+	assert.NotContains(t, htmlDoc.doc.Find(".user.profile").Text(), "user1@noreply.example.org")
 	assert.Contains(t, htmlDoc.doc.Find(".user.profile").Text(), "user1@example.com")
 
-	// user2 can see own hidden email
+	// user2 cannot see own hidden email, but a noreply email
 	session = loginUser(t, "user2")
 	req = NewRequest(t, "GET", "/user2")
 	resp = session.MakeRequest(t, req, http.StatusOK)
 	htmlDoc = NewHTMLParser(t, resp.Body)
-	assert.Contains(t, htmlDoc.doc.Find(".user.profile").Text(), "user2@example.com")
+	assert.Contains(t, htmlDoc.doc.Find(".user.profile").Text(), "user2@noreply.example.org")
+	assert.NotContains(t, htmlDoc.doc.Find(".user.profile").Text(), "user2@example.com")
 
 	setting.UI.ShowUserEmail = false
 
-	// user1 can see own (now hidden) email
+	// user1 can see own noreply email (email hidden)
```

Comment:
The current tests below suggest the opposite of this, ie. that user still sees (`.Contains`) their own e‐mail (`user1@example.com`) and that they _don’t_ see (`.NotContains`) their noreply e‐mail (`user[12]@noreply.example.org`)… which way is it supposed to be?

Reply:
ACK, this is a problem.

Reply:
Investigating possible logic error when ShowUserEmail is disabled in the config

Reply:
I seem to have gotten super confused about the intention of `ShowUserEmail`, which, according to the Config Cheat sheet, only controls whether an email will show up on the Explore page or not. With that being said, I am not sure why the config switch is there to begin with and what it actually controls, as there is no Explore stuff being changed.

Reply:
OK, so the reason why I got confused by the tests was because the test was using `ShowUserEmail` despite the setting only really concerning whether the email shows up in the Explore page, according to the Config Cheat Sheet. There was no logic error, but I got confused as to what the test actually did because of the comments (that I previously rephrased without checking the content) and the comments showed that.

I previously touched that file, but the config switch seems to have existed before my time here. I will fix that mistake in the test and it should be OK.

Reply:
Not sure if this investigating is still on-going, but the current/new comment still doesn’t seem to match what the tests are testing. (If it’s still being investigated, then disregard this comment!)